src/DateTimeRange.jsx

```diff
@@ -18,6 +18,13 @@ var DateTimeRange = React.createClass({
     };
   },
 
+  getInitialState: function() {
+    return {
+      overrideEndDate: (this.props.end === undefined ? false : true),
```

Comment:
Does this need to be 2 separate bits of state? We want to know whether to allow automatic movement of the end date, but probably don't care _why_ it's no longer moving.